aten/src/ATen/native/TensorShape.cpp

```diff
@@ -1567,6 +1567,11 @@ Tensor reshape_symint(const Tensor& self, c10::SymIntArrayRef proposed_shape) {
   if (self.is_sparse()) {
     AT_ERROR("reshape is not implemented for sparse tensors");
   }
+
+  if (self.is_contiguous() && !self.is_mkldnn()) {
```

Comment:
Are we eventually going to have to handle short-circuiting for channels last here too?

Reply:
I don't plan on adding it unless it shows up as a thing.